packages/browser/lib/main.ts

```diff
@@ -364,35 +364,46 @@ export default class EvervaultClient {
         if (hiddenInput === null) {
           return;
         }
+
         const parentForm = findParentOfInput(hiddenInput);
-        form.targetElements.forEach((field, idx) => {
-          const childToEncrypt = findChildOfForm(
-            parentForm,
-            field.elementType,
-            field.elementName
-          );
-          childToEncrypt.removeAttribute("name");
-
-          const hiddenField = document.createElement(field.elementType);
-          hiddenField.setAttribute("type", "hidden");
-          hiddenField.setAttribute("name", field.elementName);
-          hiddenField.setAttribute("id", `ev_enc_shadow_${idx}`);
-          hiddenField.style.visibility = "hidden";
-          parentForm.appendChild(hiddenField);
-
-          childToEncrypt.addEventListener("input", (event) => {
-            const target = event.target as HTMLTextAreaElement;
-            if (target?.value) {
-              this.encrypt(target.value)
-                .then((encryptedValue) => {
-                  // @ts-expect-error explict cast is needed for more then a textarea
-                  hiddenField.value = encryptedValue;
-                })
-                .catch((_) => {
-                  console.error("Error encrypting form value");
-                });
-            }
+        if (parentForm === null) {
+          return;
+        }
+
+        const actionAttr = parentForm.getAttribute("action");
+        if (actionAttr === null) {
+          return;
+        }
+        parentForm.removeAttribute("action");
+
+        const formSubmitButton = parentForm.querySelector("[type='submit']");
+        formSubmitButton?.addEventListener("click", (event) => {
+          event.preventDefault();
+          const mutations: Promise<string>[] = [];
+          form.targetElements.forEach((field) => {
+            const childToEncrypt = findChildOfForm(
+              parentForm,
+              field.elementType,
+              field.elementName
+            ) as HTMLInputElement | HTMLTextAreaElement;
+            mutations.push(
+              new Promise((resolve, reject) => {
+                this.encrypt(childToEncrypt.value)
+                  .then((encValue) => {
+                    resolve((childToEncrypt.value = encValue));
+                  })
+                  .catch((err) => {
+                    reject(err);
+                  });
+              })
+            );
           });
+          Promise.all(mutations)
+            .then(() => {
+              parentForm.setAttribute("action", actionAttr);
+              parentForm.submit();
+            })
+            .catch(() => console.error("Error encrypting form value"));
```

Comment:
any point in logging the error here for context?

Reply:
Sure!

Reply:
Sure